modules/core/src/viewports/web-mercator-viewport.js

```diff
@@ -69,14 +71,16 @@ export default class WebMercatorViewport extends Viewport {
     // TODO - just throw an Error instead?
     altitude = Math.max(0.75, altitude);
 
-    const {fov, aspect, focalDistance, near, far} = getProjectionParameters({
+    const fovy = altitudeToFovy(projectionMatrix ? projectionMatrix[5] / 2 : altitude);
+    const {aspect, fov: fovyRadians, near, far} = getProjectionParameters({
```

Comment:
Only run this if `projectionMatrix` is not supplied?

```js
let fovy;
if (projectionMatrix) {
  fovy = altitudeToFovy(projectionMatrix[5] / 2);
} else {
  fovy = altitudeToFovy(altitude);
  projectionMatrix = getProjectionMatrix(...);
}
```

Reply:
I've done this refactor, but with a slightly different approach so as to support the orthographic code path